resources/intro-to-observables.md

```diff
@@ -0,0 +1,123 @@
+
+Observables are now included in the current version of JavaScript (as of ES7) and Angular now has them baked in using a third-party library called RxJS (Reactive Extensions). In Angular, Observables are used for its event system and HTTP client service. In addition, we're starting to see more React projects use RxJS to take advantage of Observables to manage streams.
+
+## What is an Observable?
+
+Observables originate from something called the **observer design pattern**. 
+
+> The observer pattern is a software design pattern in which an object, called the **subject**, maintains a list of its dependents, called **observers**, and notifies them automatically of any state changes, usually by calling one of their methods. ([source](https://en.wikipedia.org/wiki/Observer_pattern)) 
+
+Ok, so that definition is pretty wordy and technical. 
+
+In simple terms, the pattern involves an object (called an observable) that will share (or emit) data as it arrives over time. Throughout your code, many different objects (called observers) can sign up to be notified when that data arrives (called subscribing). So, whenever our observable recieves its data, it will broadcast that information out and the observers will be listening for it.  
+
+Some key points to note about the pattern:
+
+- You can have one observable and many observers (a one-to-many dependancy). That is, one broadcaster and an unlimited number of recipients.
+- When that observable changes, all observers should be updated/notified immediately and automatically
+- It is the observer's responsibility to register and unregister themselves from the observable (to be notified of changes) and to update their own state when notified.
+
+## So... an Observable?
+
+Think of an observable as a box that you create to hold data from a service or function whose data or items will arrive over time. 
+
+As the data arrives, the box notifies anyone that has requested they be updated and it does so in the way they've asked. (Note: once new data arrives, the previous data no longer exists in the box. It's up to the listener to maintain the previous data if needed.)  
+
+## Basic structure
+
+In its simpliest form, an Observable is a function that accepts an object (an observer) with next, error and complete methods on it. 
+
+When you create an observable, you call its constructor and pass in a function. That function takes in an observer as a parameter and its body contains instructions for the observable to know when to call each of the observer's methods. 
+
+For example ([source](https://medium.com/@benlesh/learning-observable-by-building-observable-d5da57405d87))
+
+```javascript
+const this.data = Observable.create(observer => {
+    const datasource = new DataSource();
+    datasource.ondata = (e) => observer.next(e);
+    datasource.onerror = (err) => observer.error(err);
+    datasource.oncomplete = () => observer.complete();
+    return () => {
+        datasource.destroy();
+    };
+});
+```
```

Comment:
I have mixed feelings about this above example. I like that you can clearly see how when the datasource's events happen, it'll call the appropriate observer function. But two concerns:
1) datasource isn't really a thing but it looks like it's a built-in official thing
2) usually, we use timeouts in the examples (see the jsbin link below) but I thought that might be harder to read as a first example: `setTimeout(() => observer.next('hello'), 2000);` 

Thoughts?